novawallet/Modules/Vote/Governance/Referendums/ReferendumsInteractor.swift

```diff
@@ -352,3 +357,11 @@ final class ReferendumsInteractor: AnyProviderAutoCleaning, AnyCancellableCleani
         operationQueue.addOperations(votingWrapper.allOperations, waitUntilFinished: false)
     }
 }
+
+extension ReferendumsInteractor: EventVisitorProtocol {
+    func processNetworkEnableChanged(event: NetworkEnabledChanged) {
+         if let option = governanceState.settings.value {
```

Comment:
The same is here. First, we need to check that the current option is changed chain. Then we need setup settings and get the new chain.